app/webpacker/components/ContactEditProfilePage/EditProfileForm.jsx

```diff
@@ -0,0 +1,185 @@
+        />
+      )}
+      {!loggedInUserWcaId && (
+        <Form.Field
+          control={IdWcaSearch}
+          name="wcaId"
+          label={i18n.t('page.contact_edit_profile.form.wca_id.label')}
+          value={customWcaId}
+          onChange={(e, { value }) => setCustomWcaId(value)}
+          disabled={customWcaId}
+          multiple={false}
+          model={SEARCH_MODELS.person}
+          required
+        />
+      )}
+      {wcaId && (
+        <>
+          <Form.Input
+            label={i18n.t('activerecord.attributes.user.name')}
+            name="name"
+            value={editedProfileDetails?.name}
+            onChange={handleFormChange}
+          />
+          <Form.Select
+            options={countryOptions}
+            label={i18n.t('activerecord.attributes.user.country_iso2')}
+            name="country_iso2"
+            search
+            value={editedProfileDetails?.country_iso2}
+            onChange={handleFormChange}
+          />
+          <Form.Select
+            options={genderOptions}
+            label={i18n.t('activerecord.attributes.user.gender')}
+            name="gender"
+            value={editedProfileDetails?.gender}
+            onChange={handleFormChange}
+          />
+          <Form.Field
+            label={i18n.t('activerecord.attributes.user.dob')}
+            name="dob"
+            control={UtcDatePicker}
+            showYearDropdown
+            dateFormatOverride="YYYY-MM-dd"
+            dropdownMode="select"
+            isoDate={editedProfileDetails?.dob}
+            onChange={(date) => handleFormChange(null, {
+              name: 'dob',
+              value: date,
+            })}
+          />
+          <Form.TextArea
+            label={i18n.t('page.contact_edit_profile.form.edit_reason.label')}
+            name="editProfileReason"
+            required
+            value={editProfileReason}
+            onChange={handleEditProfileReaconChange}
+          />
+          <Form.Input
+            label={i18n.t('page.contact_edit_profile.form.proof_attach.label')}
+            type="file"
+            onChange={handleProofUpload}
+          />
+        </>
+      )}
+      <Form.Field>
+        <ReCAPTCHA
+          sitekey={recaptchaPublicKey}
+          // onChange is a mandatory parameter for ReCAPTCHA. According to the documentation, this
+          // is called when user successfully completes the captcha, hence we are assuming that any
+          // existing errors will be cleared when onChange is called.
+          onChange={setCaptchaValue}
+          onErrored={setCaptchaError}
+        />
+        {captchaError && (
+          <Message
+            error
+            content={i18n.t('page.contact_edit_profile.form.captcha.validation_error')}
+          />
+        )}
+      </Form.Field>
+      <Form.Button
+        type="submit"
+        disabled={
+          !editedProfileDetails || _.isEqual(editedProfileDetails, profileDetails) || !captchaValue
```

Comment:
Please put at least the `_.isEqual` check, or potentially even the whole `||` chain, into its own `useMemo` variable

Reply:
Done